### PR TITLE
[Enhancement] Organize the data directory structure on the object store by database

### PR DESCRIPTION
**Pull request by sduzh**
## Why I'm doing:
Make it easier for administrators to manage the data on the object store, for example, it is easier to count the space used by a database on the object store.

## What I'm doing:
Organize the data directory structure on the object store by database

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 1]
```diff
@@ -191,24 +191,29 @@ private FileStoreType getFileStoreType(String storageType) {
         return null;
     }
 
-    public FilePathInfo allocateFilePath(long tableId) throws DdlException {
+    private static String constructTablePath(long dbId, long tableId) {
+        return String.format("DB%d/%d", dbId, tableId);
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:195]: use lower case?

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java]: Updated

**Review by kevincai — DISMISSED** [review]

File: fe/fe-core/src/test/java/com/starrocks/lake/CreateLakeTableTest.java [hunk 2]
```diff
@@ -119,9 +119,10 @@ public void testCreateLakeTable() throws UserException {
                         "properties('replication_num' = '1');"));
         checkLakeTable("lake_test", "multi_partition_unique_key");
 
+        Database db = GlobalStateMgr.getCurrentState().getDb("lake_test");
         LakeTable table = getLakeTable("lake_test", "multi_partition_unique_key");
         String defaultFullPath = getDefaultStorageVolumeFullPath();
-        String defaultTableFullPath = String.format("%s/%s", defaultFullPath, table.getId());
+        String defaultTableFullPath = String.format("%s/DB%d/%d", defaultFullPath, db.getId(), table.getId());
```
Comment (kevincai) [inline fe/fe-core/src/test/java/com/starrocks/lake/CreateLakeTableTest.java:125]: expect UT fail here lol

Comment (sduzh) [reply fe/fe-core/src/test/java/com/starrocks/lake/CreateLakeTableTest.java]: Fixed

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**